src/openvic-simulation/history/CountryHistory.cpp

```diff
@@ -46,9 +50,30 @@ bool CountryHistoryMap::_load_history_entry(
 					return true;
 				})(value);
 			}
-			// TODO: technologies & inventions
+
+			Technology const* technology = technology_manager.get_technology_by_identifier(key);
+			if (technology != nullptr) {
+				uint8_t flag = -1;
+				if (expect_uint(assign_variable_callback(flag))(value)) {
+					if (flag == 1) return entry.technologies.emplace(technology, true).second;
+					else if (flag == 0) return entry.technologies.emplace(technology, false).second;
+					else {
+						Logger::warning("Refusing to load country history technology with non-boolean value: ", key, ", ", flag);
+						return true;
+					}
+				} else return false;
```

Comment:
You can do this similarly to how you've done inventions by using `expect_int_bool`, which converts 0->false, 1->true.

Reply:
Oh, i just assumed it would only work with actual bool types